### PR TITLE
Passport

### PR DESCRIPTION
**Pull request by pin3da**
Added github authentication

### CODE REVIEW

File: browser/main.js [hunk 1]
```diff
@@ -13,97 +13,86 @@ var Problems = require('./components/problems.js');
...
 var SubmissionForm = require('./components/submission_form.js');
 var SubmissionList = require('./components/submission_list.js');
+var TestAuth = require('./components/test_auth.js');
 
-window.backendAddress = 'http://127.0.0.1:8080/';
+var config = require('../config/development.js');
+window.backendAddress = config.backend.url;
 
 page('/contests/new', function() {
-  //document.addEventListener('DOMContentLoaded', function() {
```
Comment (leiverandres) [inline browser/main.js]: Why did you remove all this listeners? was not needed?

Comment (pin3da) [inline browser/main.js]: I just deleted commented lines :unamused:

**Comment by leiverandres** [comment]
I have a problem, when i click log in with github, it takes me to a 404 error page.

**Comment by pin3da** [comment]
@leiverandres there are some possible reasons:

- you need to configure your dns.
- you need to set up a reverse proxy in order to redirect the petitions.
- you need to get your own keys from github.

All are explained here:

https://github.com/in-silico/utpjudge

**Comment by caal-15** [comment]
All is working fine once configuration is done properly
